### PR TITLE
Group 1 - Exercise 3

### PR DESCRIPTION
**Pull request by zopad**
Group 1

### CODE REVIEW

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 1]
```diff
@@ -1,19 +1,53 @@
...
 	public void run(final File csvFile) throws Exception {
-		// TODO: Start coding here
-	}
+        this.csvFile = parseFile(csvFile);
+        fillUi();
+    }
+    
+    private CsvFile parseFile(final File csvFile) throws IOException {
+        List<String> lines = Files.readAllLines(csvFile.toPath());
+        if (lines.size() > 1) {
```
Comment (corebonts) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:26]: I think a CSV file is valid even if it only contains a header or even if it's empty.

**Comment by zopad** [comment]
Restrictions:

3. No primitive types
6. Keep entities small
7. Not more than 2 fields per class

(but we didn't plan with these in mind :D )

File: src/main/java/de/cas/mse/exercise/csv/CsvFile.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public List<String> getHeaders() {
+        return headers;
+    }
+
+    public List<List<String>> getRows() {
+        return rows;
+    }
+
+    @Override
+    public List<List<String>> filter(int col, String word) {
```
Comment (corebonts) [inline src/main/java/de/cas/mse/exercise/csv/CsvFile.java:27]: no primitive types?

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 3]
```diff
@@ -1,19 +1,53 @@
...
+            List<String> headers = getCells(lines.get(0));
+            List<List<String>> rows = lines.stream().skip(1).map(line -> getCells(line)).collect(Collectors.toList());
+            return new CsvFile(headers, rows);
+        } else {
+            throw new IllegalArgumentException("The csv file was corrupt or empty.");
+        }
+    }
+
+    private List<String> getCells(String commaSeparatedLine) {
```
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:35]: I like `commaSeparatedLine`

File: src/main/java/de/cas/mse/exercise/csv/CsvFile.java (hunk 2 above)
Comment (zopad) [reply src/main/java/de/cas/mse/exercise/csv/CsvFile.java:27]: missed that, sry

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java (hunk 1 above)
Comment (zopad) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java:26]: you're right

**Pull request merged**